shardtree/src/prunable.rs

```diff
@@ -905,6 +905,35 @@ impl<H: Hashable + Clone + PartialEq> LocatedPrunableTree<H> {
             }
         }
 
+        /// Combines the given subtree with an empty sibling node to obtain the next level
+        /// subtree.
+        ///
+        /// `expect_left_child` is set to a constant at each callsite, to ensure that this
+        /// function is only called on either the left-most or right-most subtree.
+        fn combine_with_empty<H: Hashable + Clone + PartialEq>(
+            root: LocatedPrunableTree<H>,
+            expect_left_child: bool,
+            incomplete: &mut Vec<IncompleteAt>,
+            contains_marked: bool,
+            prune_below: Level,
+        ) -> LocatedPrunableTree<H> {
+            let sibling_addr = root.root_addr.sibling();
```

Comment:
Maybe panic earlier:
```suggestion
            assert_eq!(expect_left_child, root.root_addr.is_left_child);
            let sibling_addr = root.root_addr.sibling();
```